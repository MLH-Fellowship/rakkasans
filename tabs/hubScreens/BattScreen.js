import React from "react";
import { StyleSheet, View } from "react-native";
import { useNavigation } from "@react-navigation/native";
import Colors from "../../constants/Colors";
import BattalionButton from "../../components/BattalionButton";

export default function BattScreen() {
  const navigation = useNavigation();
  return (
    <View style={styles.container}>
      <View style={styles.buttonPair}>
        <BattalionButton
          onPress={() =>
            navigation.navigate("Battalion", {
              name: "HHC",
              image: require("../../assets/images/battalions/HHC.png"),
            })
          }
          image={require("../../assets/images/battalions/HHC.png")}
          iconWidth={65}
          iconHeight={70}
          text="HHC"
        />
        <BattalionButton
          onPress={() =>
            navigation.navigate("Battalion", {
              name: "1-187 Leader BN",
              image: require("../../assets/images/battalions/1-187.png"),
            })
          }
          image={require("../../assets/images/battalions/1-187.png")}
          iconWidth={65}
          iconHeight={65}
          text="1-187"
        />
      </View>

      <View style={styles.buttonPair}>
        <BattalionButton
          onPress={() =>
            navigation.navigate("Battalion", {
              name: "2-506 White Curahee",
              image: require("../../assets/images/battalions/2-506.png"),
            })
          }
          image={require("../../assets/images/battalions/2-506.png")}
          iconWidth={65}
          iconHeight={65}
          text="2-506"
        />
        <BattalionButton
          onPress={() =>
            navigation.navigate("Battalion", {
              name: "3-187 Iron BN",
              image: require("../../assets/images/battalions/1-187.png"),
            })
          }
          image={require("../../assets/images/battalions/1-187.png")}
          iconWidth={65}
          iconHeight={65}
          text="3-387"
        />
      </View>

      <View style={styles.buttonPair}>
        <BattalionButton
          onPress={() =>
            navigation.navigate("Battalion", {
              name: "1-33 War",
              image: require("../../assets/images/battalions/1-33.png"),
            })
          }
          image={require("../../assets/images/battalions/1-33.png")}
          iconWidth={70}
          iconHeight={70}
          text="1-33"
        />
        <BattalionButton
          onPress={() =>
            navigation.navigate("Battalion", {
              name: "3-320th FA BN",
              image: require("../../assets/images/battalions/3-320th.png"),
            })
          }
          image={require("../../assets/images/battalions/3-320th.png")}
          iconWidth={65}
          iconHeight={65}
          text="3-320th"
        />
      </View>

      <View style={styles.buttonPair}>
        <BattalionButton
          onPress={() =>
            navigation.navigate("Battalion", {
              name: "21 BEB",
              image: require("../../assets/images/battalions/21BEB.png"),
            })
          }
          image={require("../../assets/images/battalions/21BEB.png")}
          iconWidth={65}
          iconHeight={65}
          text="21 BEB"
        />
        <BattalionButton
          onPress={() =>
            navigation.navigate("Battalion", {
              name: "626 BSB",
              image: require("../../assets/images/battalions/626BSB.png"),
            })
          }
          image={require("../../assets/images/battalions/626BSB.png")}
          iconWidth={65}
          iconHeight={65}
          text="626 BSB"
        />
      </View>
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 5,
    backgroundColor: Colors.white,
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  buttonPair: {
    width: "100%",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-evenly",
    paddingBottom: 20,
  },
});
