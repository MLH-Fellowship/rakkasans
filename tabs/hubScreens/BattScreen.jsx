import React from 'react';
import { StyleSheet, View } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Colors from '../../constants/Colors';
import BattalionButton from '../../components/BattalionButton';

import HHCPic from '../../assets/images/battalions/HHC.png';
import Batt1187Pic from '../../assets/images/battalions/1-187.png';
import Batt2506Pic from '../../assets/images/battalions/2-506.png';
import Batt133Pic from '../../assets/images/battalions/1-33.png';
import Batt3320Pic from '../../assets/images/battalions/3-320th.png';
import Batt21Pic from '../../assets/images/battalions/21BEB.png';
import Batt626Pic from '../../assets/images/battalions/626BSB.png';

export default function BattScreen() {
  const navigation = useNavigation();
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: 5,
      backgroundColor: Colors.white,
      width: '100%',
      alignItems: 'center',
      justifyContent: 'center',
    },
    buttonPair: {
      width: '100%',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-evenly',
      paddingBottom: 20,
    },
  });
  return (
    <View style={styles.container}>
      <View style={styles.buttonPair}>
        <BattalionButton
          onPress={() => navigation.navigate('Battalion', {
            name: 'HHC',
            image: HHCPic,
          })}
          image={HHCPic}
          iconWidth={65}
          iconHeight={70}
          text="HHC"
        />
        <BattalionButton
          onPress={() => navigation.navigate('Battalion', {
            name: '1-187 Leader BN',
            image: Batt1187Pic,
          })}
          image={Batt1187Pic}
          iconWidth={65}
          iconHeight={65}
          text="1-187"
        />
      </View>

      <View style={styles.buttonPair}>
        <BattalionButton
          onPress={() => navigation.navigate('Battalion', {
            name: '2-506 White Curahee',
            image: Batt2506Pic,
          })}
          image={Batt2506Pic}
          iconWidth={65}
          iconHeight={65}
          text="2-506"
        />
        <BattalionButton
          onPress={() => navigation.navigate('Battalion', {
            name: '3-187 Iron BN',
            image: Batt1187Pic,
          })}
          image={Batt1187Pic}
          iconWidth={65}
          iconHeight={65}
          text="3-387"
        />
      </View>

      <View style={styles.buttonPair}>
        <BattalionButton
          onPress={() => navigation.navigate('Battalion', {
            name: '1-33 War',
            image: Batt133Pic,
          })}
          image={Batt133Pic}
          iconWidth={70}
          iconHeight={70}
          text="1-33"
        />
        <BattalionButton
          onPress={() => navigation.navigate('Battalion', {
            name: '3-320th FA BN',
            image: Batt3320Pic,
          })}
          image={Batt3320Pic}
          iconWidth={65}
          iconHeight={65}
          text="3-320th"
        />
      </View>

      <View style={styles.buttonPair}>
        <BattalionButton
          onPress={() => navigation.navigate('Battalion', {
            name: '21 BEB',
            image: Batt21Pic,
          })}
          image={Batt21Pic}
          iconWidth={65}
          iconHeight={65}
          text="21 BEB"
        />
        <BattalionButton
          onPress={() => navigation.navigate('Battalion', {
            name: '626 BSB',
            image: Batt626Pic,
          })}
          image={Batt626Pic}
          iconWidth={65}
          iconHeight={65}
          text="626 BSB"
        />
      </View>
    </View>
  );
}
