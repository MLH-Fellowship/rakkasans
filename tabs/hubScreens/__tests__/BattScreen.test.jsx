import React from 'react';
import renderer from 'react-test-renderer';

import BattScreen from '../BattScreen';

jest.useFakeTimers();
jest.mock('@react-navigation/native');

it('renders correctly', () => {
  const tree = renderer.create(<BattScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
