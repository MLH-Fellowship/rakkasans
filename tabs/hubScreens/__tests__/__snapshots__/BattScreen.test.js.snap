// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
      "paddingTop": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 85,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 1,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
            "width": 85,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "width": 65,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "paddingTop": 8,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        HHC
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 85,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 1,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
            "width": 85,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 65,
              "width": 65,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "paddingTop": 8,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        1-187
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 85,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 1,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
            "width": 85,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 65,
              "width": 65,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "paddingTop": 8,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        2-506
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 85,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 1,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
            "width": 85,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 65,
              "width": 65,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "paddingTop": 8,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        3-387
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 85,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 1,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
            "width": 85,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "paddingTop": 8,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        1-33
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 85,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 1,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
            "width": 85,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 65,
              "width": 65,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "paddingTop": 8,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        3-320th
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 85,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 1,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
            "width": 85,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 65,
              "width": 65,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "paddingTop": 8,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        21 BEB
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 85,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 1,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
            "width": 85,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 65,
              "width": 65,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "paddingTop": 8,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        626 BSB
      </Text>
    </View>
  </View>
</View>
`;
